src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -192,24 +192,37 @@ public UnfilteredRowIterator getPartition(PrimaryKey key, ReadExecutionControlle
         if (key == null)
             throw new IllegalArgumentException("non-null key required");
 
+        SinglePartitionReadCommand partition = getPartitionReadCommand(key, executionController);
+        return initPartitionReadCommand(partition, executionController);
+    }
+
+    public SinglePartitionReadCommand getPartitionReadCommand(PrimaryKey key, ReadExecutionController executionController)
+    {
+        if (key == null)
+            throw new IllegalArgumentException("non-null key required");
+
+        return SinglePartitionReadCommand.create(cfs.metadata(),
+                                                 command.nowInSec(),
+                                                 command.columnFilter(),
+                                                 RowFilter.NONE,
+                                                 DataLimits.NONE,
+                                                 key.partitionKey(),
+                                                 makeFilter(key));
+    }
+
+    public UnfilteredRowIterator initPartitionReadCommand(SinglePartitionReadCommand command, ReadExecutionController executionController)
```

Comment:
is this named correctly?  looks like it's processing the command and not just initializing it